core/src/main/scala/akka/kafka/internal/BaseSingleSourceLogic.scala

```diff
@@ -52,11 +49,12 @@ private[kafka] abstract class BaseSingleSourceLogic[K, V, Msg](
           buffer = msg.messages
         }
         pump()
-      case (_, Failure(e)) =>
+      case (_, Status.Failure(e)) =>
         failStage(e)
       case (_, Terminated(ref)) if ref == consumerActor =>
         failStage(new ConsumerFailed())
     }
+    consumerActor = createConsumerActor()
     sourceActor.watch(consumerActor)
```

Comment:
`watch` returns the same `ActorRef` that it gets. So it is possible to rewrite this with:

```scala
consumerActor = sourceActor.watch(createConsumerActor())
```

Reply:
No, it is `Unit` typed.

Reply:
Ahh, `sourceActor` is a stage sctor. Well that is unfortunate that the API differs.